test/config_cat_test.exs

```diff
@@ -144,12 +145,27 @@ defmodule ConfigCatTest do
         start_config_cat(sdk_key, base_url: base_url, fetch_policy: FetchPolicy.manual())
 
       APIMock
-      |> expect(:get, fn ^url, _headers ->
+      |> expect(:get, fn ^url, _headers, _options ->
         {:ok, %Response{status_code: 200, body: %{}}}
       end)
 
       :ok = ConfigCat.force_refresh(client)
     end
+
+    test "sends proper http proxy options" do
+      {:ok, client} = start_config_cat("SDK_KEY", fetch_policy: FetchPolicy.manual(), http_proxy: "https://myproxy.com")
+
+      response = %Response{status_code: 200, body: %{}}
+
+      APIMock
+      |> stub(:get, fn _url, _headers, options ->
+        assert options == [proxy: "https://myproxy.com"]
```

Comment:
Again, this can be more concisely written as `stub(:get, fn _url, _headers, [proxy: "https://myproxy.com"]`.

When there is an `assert` in a stub, if the code under test is wrong, the stub might not ever get called, and so the assertion would never fire.  As I said above, I was forced to do this for testing the headers, but I try to avoid that pattern as much as possible.

Reply:
Good catch! Done! 👍